pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/scheduler/UnloadScheduler.java

```diff
@@ -131,9 +131,11 @@ public synchronized void execute() {
                 List<CompletableFuture<Void>> futures = new ArrayList<>();
                 unloadDecision.getUnloads().forEach((broker, unload) -> {
                     log.info("[{}] Unloading bundle: {}", namespaceUnloadStrategy.getClass().getSimpleName(), unload);
-                    futures.add(channel.publishUnloadEventAsync(unload).thenAccept(__ -> {
-                        recentlyUnloadedBundles.put(unload.serviceUnit(), System.currentTimeMillis());
-                        recentlyUnloadedBrokers.put(unload.sourceBroker(), System.currentTimeMillis());
+                    futures.add(channel.publishUnloadEventAndWaitUnloadComplete(
+                            unload, conf.getNamespaceBundleUnloadingTimeoutMs(), TimeUnit.MILLISECONDS)
+                            .thenAccept(__ -> {
+                                recentlyUnloadedBundles.put(unload.serviceUnit(), System.currentTimeMillis());
+                                recentlyUnloadedBrokers.put(unload.sourceBroker(), System.currentTimeMillis());
```

Comment:
Please consider UnloadMetrics if you can update it in this PR. I am fine with another PR for this.